node/src/daemon/setup_reporter.rs

```diff
@@ -78,10 +85,9 @@ impl SetupReporter for SetupReporterReal {
                     blanked_out_former_values.insert(v.name.clone(), former_value);
                 };
             });
-        //we had troubles at an attempt to blank out this parameter on an error resulting in a diverging chain from the data_dir
-        if blanked_out_former_values.get("chain").is_some() {
-            let _ = blanked_out_former_values.remove("chain");
-        }
+        //TODO investigate this, not sure if the right way to solve the issue
+        //answers an attempt to blank out chain beyond an error resulting in chain different from data_dir
+        let _ = blanked_out_former_values.remove("chain");
```

Comment:
Seems reasonable. Would it make sense to move `blanked_out_former_values.remove("chain");` into the block above? EX:
`if let Some(former_value) = existing_setup.remove(&v.name) {
                    blanked_out_former_values.insert(v.name.clone(), former_value);
                    blanked_out_former_values.remove("chain");
                };`

Reply:
Well, the block above iterates in a loop and we would have to evaluate the expression every time, with it placed outside we know we do it just once.